### PR TITLE
Remove laravelcollective/html from frontend

### LINKED ISSUE #1710: Replace laravelcollective/html with spatie/laravel-html
laravelcollective/html's documentation states that it's been "retired". The documentation points to using spatie/laravel-html instead. 

https://spatie.be/index.php/docs/laravel-html/v3/introduction

This issue is safe to defer to 7.1, as there might be considerable rework required to implement the new package.

### PR DESCRIPTION
**Pull request by arthurpar06**
As you may already know, `laravelcollective/html` will not be compatible with Laravel 11. This PR is an attempt to start preparing for Laravel 11 by removing it from the frontend and reverting to good old pure HTML.

I have tested all the forms that this PR modifies, and I did not notice any bugs.

Note:
- The CSRF token is not added by default when creating HTML forms, so you need to remember to add `@csrf` each time.
- For the values, it's the same, you need to test the existence of the field and set the value (if you are in edit mode). To retrieve the value sent in the returned form, in case there is a validation error, you can use `old('field_name')`.

Closes #1710

### CODE REVIEW

**Comment by arthurpar06** [comment]
Closes #1708

**Comment by FatihKoz** [comment]
For backward compatibility, we need to keep the package in for some defined time... Like maybe with a notice and 1-2 months, because if we remove the package directly it will break up lots of custom themes and people's install will start failing over and there 😄 

I can work on my theme and provide an update without its features + plain html stuff but not everyone is that quick to follow.

Just my two cents 😉

**Comment by arthurpar06** [comment]
I completely agree, that's why this PR doesn't remove it.
I just don't use it in the frontend anymore. This way, when we upgrade to Laravel 11, we won't have to replace all the `Form::` in the frontend, just remove the package.

File: resources/views/layouts/default/auth/register.blade.php [hunk 1]
```diff
@@ -133,8 +152,7 @@
             <tr>
               <td>
                 <div class="input-group form-group-no-border">
-                  {{ Form::hidden('opt_in', 0, false) }}
-                  {{ Form::checkbox('opt_in', 1, null) }}
+                  <input type="checkbox" name="opt_in" id="opt_in" />
```
Comment (FatihKoz) [inline resources/views/layouts/default/auth/register.blade.php:155]:
```html
<input type="hidden" name="opt_in" id="opt_in" value="0">
<input type="checkbox" name="opt_in" id="opt_in" value="1">
```

Reason: When a checkbox is not check it does not get sent by the form, so to reset a selection, there needs to be a hidden field just before the main checkbox.

Ref: HTML Forms, Checkbox, Sending Data

This should be done for all checkboxes specially where editing is possible (or the backend code of these forms should be updated, to check if the checkbox value is received or not, then setting 0 / false etc. as the default value)

Comment (arthurpar06) [reply resources/views/layouts/default/auth/register.blade.php:155]: Thanks

**Comment by FatihKoz** [comment]
id's should be unique (don't mind my example), so it should be used only once, preferably if needed setting it only for the checkbox is ok.

And is this the only checkbox ? It think we have another one for accepting the TOC too.

PS: Contrary to id's, same name can be used multiple times, last one will be sent along with the form.

File: resources/views/layouts/default/auth/register.blade.php [hunk 2]
```diff
@@ -6,54 +6,74 @@
...
+            </select>
           </div>
           @if ($errors->has('country'))
             <p class="text-danger">{{ $errors->first('country') }}</p>
           @endif
 
           <label for="timezone" class="control-label">@lang('common.timezone')</label>
           <div class="input-group form-group-no-border {{ $errors->has('timezone') ? 'has-danger' : '' }}">
-            {{ Form::select('timezone', $timezones, null, ['id'=>'timezone', 'class' => 'form-control select2' ]) }}
+            <select name="timezone" id="timezone" class="form-control select2">}
```
Comment (FatihKoz) [inline resources/views/layouts/default/auth/register.blade.php:68]: There is an extra `}` at the end of this line 😉

**Comment by arthurpar06** [comment]
> id's should be unique (don't mind my example), so it should be used only once, preferably if needed setting it only for the checkbox is ok.

Oh yes sorry I'll fix that.


> And is this the only checkbox ? It think we have another one for accepting the TOC too.

Yes but we don't send the `toc_accepted` to the controller, we just want it to be checked so that the js enables the submit button

**Comment by FatihKoz** [comment]
I suggest doing another overall check cause I saw some tiny typo errors over and there... Will try to point them out with comments.

File: resources/views/layouts/default/pireps/fields.blade.php [hunk 3]
```diff
@@ -96,46 +112,55 @@
...
-                    'step' => '0.01',
-                    'readonly' => (!empty($pirep) && $pirep->read_only),
-                    ]) }}
+                <input type="number"
+                       name="level"
+                       id="level"
+                       class="form-control"
+                       @if(!empty($pirep) && $pirep->read_only) readonly @endif
+                       min="0"
+                       step="0.0.1"
```
Comment (FatihKoz) [inline resources/views/layouts/default/pireps/fields.blade.php:161]: step="0.01"

**Comment by FatihKoz** [comment]
Generally I am wondering about `select2` 'cause as far as I know it does NOT support `readonly` attribute, thus it needs to be `disabled` and when a form element is disabled, its data is not sent with the request. 

I had to implement some logic in my theme and modules (hidden fields and disable together).

So if `select2` is still not proving `readonly` support, this may break some of the checks v7 default theme needs 😉

**Comment by arthurpar06** [comment]
> Generally I am wondering about `select2` 'cause as far as I know it does NOT support `readonly` attribute, thus it needs to be `disabled` and when a form element is disabled, its data is not sent with the request.
> 
> I had to implement some logic in my theme and modules (hidden fields and disable together).
> 
> So if `select2` is still not proving `readonly` support, this may break some of the checks v7 default theme needs 😉

Fair point, I'll try to look at it tomorrow

**Comment by FatihKoz** [comment]
I updated all my addons, removed laravelcollective/html usage completely. So dispo side ok with v3.6.0 and up 😉 

When we merge this for default theme and any admin templates, all will be ready.

**Comment by arthurpar06** [comment]
I just reviewed everything again, fixed some bugs here and there, I think there are not many left now (if there are any).

I still don't remove the package from the `composer.json` for the admin part. Since the upgrade to Laravel 11 won't be done before the release, I don't think we should remove this package from the admin interface because after the release we will start reviewing the filament project (which is ready) and then we will remove this package by replacing the admin interface.

**Comment by FatihKoz** [comment]
I see you removed the readonly attributes from select2 dropdowns, are you going to disable them? Or what will be the solution for readonly fields ?

**Comment by arthurpar06** [comment]
Actually, they were unnecessary, we never enter this if statement. If you look at the code a bit more broadly, you'll see that these selects are within an if/else block, and the select is only displayed if the field is not readonly. If it's readonly, then only a label and text are displayed.

**Comment by FatihKoz** [comment]
Ok then :)

File: resources/views/layouts/default/auth/register.blade.php [hunk 4]
```diff
@@ -6,54 +6,74 @@
     <div class="col-sm-3"></div>
     <div class="col-sm-6">
 
-      {{ Form::open(['url' => '/register', 'class' => 'form-signin']) }}
-
+      <form method="post" action="{{ url('/register') }}" class="form-signin">
```
Comment (FatihKoz) [inline resources/views/layouts/default/auth/register.blade.php:9]: Did you tested this @arthurpar06 ?

I have reports about this form returning back to the same page without registering users, trying to replicate it on local and dev environments.

Comment (arthurpar06) [reply resources/views/layouts/default/auth/register.blade.php:9]: As discussed on discord it was a typo error in your theme right ?

Comment (FatihKoz) [reply resources/views/layouts/default/auth/register.blade.php:9]: Exactly, fixed.

**Comment by nabeelio** [comment]
This one makes me nervous. Is there another form helper that can be used? And what about the admin?

**Comment by arthurpar06** [comment]
There is https://github.com/spatie/laravel-html which is the recommended alternative, but I don't feel like it really brings anything compared to pure HTML.

For the admin I didn't take care of it because it takes a lot of time (there are many more forms) and if we plan to migrate to Filament after the release and before upgrading to Laravel 11 I didn't see the point of doing it.

**Comment by ArthurHetem** [comment]
@arthurpar06 i think we should remove the laravelHtml from the installer and updater too

**Comment by arthurpar06** [comment]
Oh yea, nice catch... @ArthurHetem do you wanna do it? Otherwise, I can take care of it, probably over the weekend

**Pull request merged**